operations/UAS_user_and_admin_topics/Troubleshoot_Broker_SSSD_Cant_Use_sssd_conf.md

```diff
@@ -0,0 +1,151 @@
+
+```
+ncn-m001:~ # kubectl get clusterrolebindings uas-default-psp -o yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+
+...
+
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: restricted-transition-net-raw-psp
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: uas
+```
+
+Notice the `roleRef` setting binds the `restricted-transition-net-raw-psp` Cluster Role. That Cluster Role uses the `restricted-transition-net-raw-psp` Pod Security Policy, which is used by several CSM services, but does not work for Broker UAIs.
+
+Assuming this configuration is present, it is necessary to create a new Pod Security Policy called `uas-default-psp`, then create a Cluster Role called `uas-default-psp` that uses the new Pod Security Policy and finally a change the Cluster Role Binding called `uas-defaul-psp` to bind the new Cluster Role to the `default` Service Account in the `uas` namespace.  Since Kubernetes does not allow simply updating the existing `uas-default-psp` Cluster Role Binding the existing one must be removed and replaced.  To remove the existing Cluster Role Binding:
+
+```
+ncn-m001:~ # kubectl delete clusterrolebindings uas-default-psp
+```
+
+Now prepare a YAML file containing the new Kubernetes objects:
+
+```
+ncn-m001:~ # cat << EOF > /tmp/uas-default-psp.yaml
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: uas-default-psp
+spec:
+  allowPrivilegeEscalation: true
+  allowedCapabilities:
+  - NET_ADMIN
+  - NET_RAW
+  allowedHostPaths:
+  - pathPrefix: /lustre
+  - pathPrefix: /root/registry
+  - pathPrefix: /lib/modules
+  - pathPrefix: /
+  - pathPrefix: /var/lib/nfsroot/nmd
+  - pathPrefix: /lus
+  - pathPrefix: /var/tmp/cps-local
+  fsGroup:
+    rule: RunAsAny
+  hostNetwork: true
+  privileged: true
+  runAsUser:
+    rule: RunAsAny
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  volumes:
+  - configMap
+  - emptyDir
+  - projected
+  - secret
+  - downwardAPI
+  - persistentVolumeClaim
+  - hostPath
+  - flexVolume
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: uas-default-psp
+rules:
+- apiGroups:
+  - policy
+  resourceNames:
+  - uas-default-psp
+  resources:
+  - podsecuritypolicies
+  verbs:
+  - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: uas-default-psp
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: uas-default-psp
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: uas
+EOF
+```
+
+Finally, apply this new configuration to Kubernetes:
+
+```
+ncn-m001:~ # kubectl apply -f /tmp/uas-default-psp.yaml 
+```
+
+After this delete and re-create the offending Broker UAI(s) and they should come up and SSSD should run properly.
+
```

Comment:
It could be helpful to include the commands required to do this.

```bash
ncn-m001# cray uas admin uais delete --uai-list UAI-NAMES
```

```bash
ncn-m001# cray uas admin uais create OPTIONS
```

Reply:
The commands to do this efficiently are likely to vary somewhat by site.  I could do something like:
```
ncn-m001# cray uas admin uais delete OPTIONS
```
```
ncn-m001# cray uas admin uais create OPTIONS
```
Though.  I will probably do that.